src/main/java/com/gluonhq/substrate/SubstrateDispatcher.java

```diff
@@ -72,20 +75,23 @@ public static void main(String[] args) throws Exception {
         config.setGraalPath(graalVM);
         config.setMainClassName(mainClass);
         config.setAppName(appName);
-        config.setJavaStaticSdkVersion(Constants.DEFAULT_JAVA_STATIC_SDK_VERSION);
         config.setJavafxStaticSdkVersion(Constants.DEFAULT_JAVAFX_STATIC_SDK_VERSION);
         config.setTarget(targetTriplet);
-        config.setUseJavaFX(useJavaFX);
         config.setUsePrismSW(usePrismSW);
         config.getIosSigningConfiguration().setSkipSigning(skipSigning);
-
+        if (staticLibs != null) {
+            config.setJavaStaticLibs(staticLibs);
+        }
+        // fail-fast: in case we're missing libraries, we don't want to start compiling
+        if (!FileDeps.setupDependencies(config)) {
```

Comment:
true. At this moment, setupDependencies can throw Exception, or it can not throw an exception and fail. That behavior needs to be documented, made consistent and tested. But that is for another PR, I think. (very important though)